Userland/Libraries/LibGfx/ImageFormats/WebPWriterLossless.cpp

```diff
@@ -75,78 +245,62 @@ static ErrorOr<void> write_VP8L_image_data(Stream& stream, Bitmap const& bitmap)
     // We can make this smarter later on.
 
     size_t const color_cache_size = 0;
-    constexpr Array alphabet_sizes = to_array<size_t>({ 256 + 24 + static_cast<size_t>(color_cache_size), 256, 256, 256, 40 }); // XXX Shared?
+    constexpr Array alphabet_sizes = to_array<size_t>({ 256 + 24 + static_cast<size_t>(color_cache_size), 256, 256, 256, 40 });
 
     // If you add support for color cache: At the moment, CanonicalCodes does not support writing more than 288 symbols.
     if (alphabet_sizes[0] > 288)
         return Error::from_string_literal("Invalid alphabet size");
 
-    bool all_pixels_are_opaque = are_all_pixels_opaque(bitmap);
-
-    PrefixCodeGroup prefix_code_group;
-    int number_of_full_channels = all_pixels_are_opaque ? 3 : 4;
-    for (int i = 0; i < number_of_full_channels; ++i) {
-        TRY(bit_stream.write_bits(0u, 1u)); // Normal code length code.
-
-        // Write code length codes.
-        constexpr int kCodeLengthCodes = 19;
-        Array<int, kCodeLengthCodes> kCodeLengthCodeOrder = { 17, 18, 0, 1, 2, 3, 4, 5, 16, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 };
-        int num_code_lengths = max(4u, find_index(kCodeLengthCodeOrder.begin(), kCodeLengthCodeOrder.end(), 8) + 1);
-
-        // "int num_code_lengths = 4 + ReadBits(4);"
-        TRY(bit_stream.write_bits(num_code_lengths - 4u, 4u));
-
-        for (int i = 0; i < num_code_lengths - 1; ++i)
-            TRY(bit_stream.write_bits(0u, 3u));
-        TRY(bit_stream.write_bits(1u, 3u));
-
-        // Write code lengths.
-        if (alphabet_sizes[i] == 256) {
-            TRY(bit_stream.write_bits(0u, 1u)); // max_symbol is alphabet_size
-        } else {
-            TRY(bit_stream.write_bits(1u, 1u)); // max_symbol is explicitly coded
-            // "int length_nbits = 2 + 2 * ReadBits(3);
-            //  int max_symbol = 2 + ReadBits(length_nbits);"
-            TRY(bit_stream.write_bits(3u, 3u));   // length_nbits = 2 + 2 * 3
-            TRY(bit_stream.write_bits(254u, 8u)); // max_symbol = 2 + 254
-        }
+    // We do use huffman coding by writing a single prefix-code-group for the entire image.
+    // FIXME: Consider using a meta-prefix image and using one prefix-code-group per tile.
 
-        auto bits_per_symbol = Array<u8, 256>::from_repeated_value(8);
-        prefix_code_group[i] = TRY(CanonicalCode::from_bytes(bits_per_symbol));
+    // FIXME: generate_huffman_lengths() currently halves a frequency cap if the maximum bit length is reached.
+    //        This has the effect of giving very frequent symbols a higher bit length than they would have otherwise.
+    //        Instead, try dividing frequencies by 2 if the maximum bit length is reached.
+    //        Then, low-frequency symbols will get a higher bit length than they would have otherwise, which might help
+    //        compressed size. (For deflate, it doesn't matter much since there blocks are 64kiB large, but for WebP
```

Comment:
Nittiest of picks: "there blocks" -> "their blocks"

Reply:
Fixed. (I also have a local branch that fixes this FIXME – `4 insertions(+), 12 deletions(-)` and reduces output of sunset-retro.webp by 17% or so, without costing runtime, but first things first I suppose.)